### DISCUSSION TITLE
Does this application has a api?

### DISCUSSION THREAD

**Discussion by Flori00123** [infinitered/nsfwjs#577] [Q&A] [answered]
Hey,

sry i am not into javascript and nodejs, but i like this libary very much, it detects very accurate the content 18+.
I have two simple questions:

Does it have like a api so that i can send post request with the images to it and it response to me with the details?
Does it also detect critial images, like things that would be posted on liveleak, like dead persons? I dont have footage to test this.

Thanks!

**Comment by GantMan** [comment]
Currently, no on API

Would that be something you'd be interested in?  Would that be something you'd be willing to pay for?   I'm trying to gauge interest in an API.

Secondly, no it doesn't detect violence or death.

**Comment by Flori00123** [comment]
If I would be paid for it? I don't know, I am speaking to you as a private person, as a company maybe, but as a private person who is currently planning a project and thinking about how to handle moderation, I would say no, but if the project succeeds yes.

Also, hentai and porn detection are very useful, for a forum or anything like that, I would also need violence and death, its good that a user doesn't have a hentai profile picture, but if they have dead people or images of the main roles of the third Reich or swastika in it, its as bad as nude pictures if you understand what I mean...

**Comment by GantMan** [comment, marked as answer]
The way it is now, you could create your own API, or even add it on device for free.  Once it becomes an API, the costs are out of our control, unless we paywall it.  I'd hate to make a public API and then have to shut it down, yanno?   But we could make an open source API that others could host... hrmmm.

**Comment by Flori00123** [comment]
I saw that some people already did it with nsfwjs, there already are apis to host them by yourself, and I think that is fine, you could make it like MongoDB, you can host the service by yourself or you could host the API for that people.

Is it planned to extend the nsfw library for more detections? Like violence and death? or do you offer another product or open-source library, which covers that?

**Comment by GantMan** [comment]
That's not currently on our roadmap.
